tensorwaves/physics/helicityformalism/kinematics.py

```diff
@@ -1,16 +1,262 @@
+        raise NotImplementedError("Currently not implemented.")
+
+    def register_invariant_mass(self, final_state: Union[tuple, list]):
+        """Register an invariant mass :math:`s`.
+
+        Args:
+            final_state: collection of particle unique id's
+
+        Return:
+            A `str` key representing the invariant mass. It can be used to
+            retrieve this invariant mass from the dataset returned by
+            :meth:`~convert`.
+        """
+        logging.debug("registering inv mass in kinematics")
+        final_state = tuple(final_state)
+        if final_state not in self._registered_inv_masses:
+            label = 'mSq'
+            for particle_uid in final_state:
+                label += '_' + str(particle_uid)
+
+            self._registered_inv_masses[final_state] = label
+        return self._registered_inv_masses[final_state]
+
+    def register_helicity_angles(self, subsystem: SubSystem):
+        r"""Register helicity angles :math:`(\theta, \phi)` of a `SubSystem`.
+
+        Args:
+            subsystem: SubSystem to which the registered angles correspond.
+
+        Return:
+            A pair of `str` keys representing the angles. They can be used to
+            retrieve the angles from the dataset returned by :meth:`~convert`.
+        """
+        logging.debug("registering helicity angles in kinematics")
+        if subsystem not in self._registered_subsystems:
+            suffix = ''
+            for final_state in subsystem.final_states:
+                suffix += '+'
+                for particle_uid in final_state:
+                    suffix += str(particle_uid) + '_'
+                suffix = suffix[:-1]
+            if subsystem.recoil_state:
+                suffix += '_vs_'
+                for particle_uid in subsystem.recoil_state:
+                    suffix += str(particle_uid) + '_'
+                suffix = suffix[:-1]
+
+            self._registered_subsystems[subsystem] = (
+                'theta' + suffix, 'phi' + suffix)
+        return self._registered_subsystems[subsystem]
+
+    def register_subsystem(self, subsystem: SubSystem):
+        r"""Register all kinematic variables of the :class:`~SubSystem`.
+
+        Args:
+            subsystem: SubSystem to which the registered kinematic variables
+                correspond.
+
+        Return:
+            A tuple of `str` keys representing the :math:`(s, \theta, \phi)`.
+            They can be used to retrieve the kinematic data from the dataset
+            returned by :meth:`~convert`.
+        """
+        state_fs = []
+        for fs_uid in subsystem.final_states:
+            state_fs += fs_uid
+        invmass_name = self.register_invariant_mass(list(set(state_fs)))
+        angle_names = self.register_helicity_angles(subsystem)
+
+        return (invmass_name,) + angle_names
+
+    def _convert_ids_to_indices(self, ids: tuple):
+        """Convert unique ids to event indices.
```

Comment:
About the docstring here: private methods are currently not rendered in the API. Add them as well?

Reply:
I would def not render them, as they are not part of the API (just some internal function). Maybe we should disable this kind of docstring in the flake8/pylint settings as well. Maybe these docstrings are helpful for developers though.

Reply:
Yeah and, in addition, they are still available through `help`.